db/c_test.c

```diff
@@ -3687,6 +3687,47 @@ int main(int argc, char** argv) {
     rocksdb_statistics_histogram_data_destroy(hist);
   }
 
+  StartPhase("wait_for_compact_options");
+  {
+    rocksdb_wait_for_compact_options_t* wco;
+    wco = rocksdb_wait_for_compact_options_create();
+
+    rocksdb_wait_for_compact_options_set_abort_on_pause(wco, 1);
+    CheckCondition(1 ==
+                   rocksdb_wait_for_compact_options_get_abort_on_pause(wco));
+
+    rocksdb_wait_for_compact_options_set_flush(wco, 1);
+    CheckCondition(1 == rocksdb_wait_for_compact_options_get_flush(wco));
+
+    rocksdb_wait_for_compact_options_set_close_db(wco, 1);
+    CheckCondition(1 == rocksdb_wait_for_compact_options_get_close_db(wco));
+
+    rocksdb_wait_for_compact_options_set_timeout(wco, 342);
+    CheckCondition(342 == rocksdb_wait_for_compact_options_get_timeout(wco));
+
+    rocksdb_wait_for_compact_options_destroy(wco);
+  }
+  StartPhase("wait_for_compact");
+  {
+    rocksdb_close(db);
```

Comment:
While the memory leak detected by ASAN test doesn't appear to be related to your change, I think you can still avoid unnecessary destroying / re-opening db here to avoid that error.

You can also add `rocksdb_wait_for_compact()` call in `CheckCompaction()` and see if that impacts any other tests.

Reply:
Thanks for the feedback, added `rocksdb_wait_for_compact()` to `CheckCompaction()` and removed reopening of db.